core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpLeonardoDAO.scala

```diff
@@ -17,11 +17,19 @@ class HttpLeonardoDAO(leonardoConfig: LeonardoConfig) extends LeonardoDAO {
     new AppsV2Api(apiClient)
   }
 
-  override def createWDSInstance(token: String, workspaceId: UUID): Unit =
-    createApp(token, workspaceId, s"wds-$workspaceId", leonardoConfig.wdsType)
+  override def createWDSInstance(token: String, workspaceId: UUID, sourceWorkspaceId: Option[UUID] = None): Unit =
+    createApp(token, workspaceId, s"wds-$workspaceId", leonardoConfig.wdsType, sourceWorkspaceId)
 
-  override def createApp(token: String, workspaceId: UUID, appName: String, appType: String): Unit = {
+  override def createApp(token: String,
+                         workspaceId: UUID,
+                         appName: String,
+                         appType: String,
+                         sourceWorkspaceId: Option[UUID]
+  ): Unit = {
     val createAppRequest = new CreateAppRequest()
+    if (sourceWorkspaceId.nonEmpty) {
+      createAppRequest.setSourceWorkspaceId(sourceWorkspaceId.get.toString)
+    }
```

Comment:
a more scala-idiomatic implementation:
```scala
    sourceWorkspaceId.foreach { sourceid =>
      createAppRequest.setSourceWorkspaceId(sourceid.toString)
    }
```
check out this thread of opinions: https://stackoverflow.com/questions/6819373/why-is-foreach-better-than-get-for-scala-options and the scaladoc for `foreach`: https://www.scala-lang.org/api/2.13.8/scala/Option.html#foreach[U](f:A=%3EU):Unit